File: Tests.Common/CompilerGenerated/New.cs
using Xunit;
using static ExpressionToString.Tests.Categories;

namespace ExpressionToString.Tests {
    public partial class CompilerGeneratedBase {
        [Fact]
        [Trait("Category", NewObject)]
        public void NamedType() => PreRunTest();

        [Fact]
        [Trait("Category", NewObject)]
        public void NamedTypeWithInitializer() => PreRunTest();

        [Fact]
        [Trait("Category", NewObject)]
        public void NamedTypeWithInitializers() => PreRunTest();

        [Fact]
        [Trait("Category", NewObject)]
        public void NamedTypeConstructorParameters() => PreRunTest();

        [Fact]
        [Trait("Category", NewObject)]
        public void NamedTypeConstructorParametersWithInitializers() => PreRunTest();

        [Fact]
        [Trait("Category", NewObject)]
        public void AnonymousType() => PreRunTest();

        [Fact]
        [Trait("Category", NewObject)]
        public void AnonymousTypeFromVariables() => PreRunTest();

        [Fact]
        [Trait("Category", NewObject)]
        public void CollectionTypeWithInitializer() => PreRunTest();

        [Fact]
        [Trait("Category", NewObject)]
        public void CollectionTypeWithMultipleElementsInitializers() => PreRunTest();

        [Fact]
        [Trait("Category", NewObject)]
        public void CollectionTypeWithSingleOrMultipleElementsInitializers() => PreRunTest();

        [Fact]
        [Trait("Category", NewObject)]
        public void MemberMemberBinding() => PreRunTest();

        [Fact]
        [Trait("Category", NewObject)]
        public void ListBinding() => PreRunTest();
    }
}

File: Tests.Common.VB/Objects/New.vb
Imports ExpressionToString.Tests.Functions

Namespace Objects
    Partial Public Module VBCompiler
        <Category(NewObject)>
        Public NamedType As Expression = Expr(Function() New Random())

        <Category(NewObject)>
        Public NamedTypeWithInitializer As Expression = Expr(Function() New Foo With {.Bar = "abcd"})

        <Category(NewObject)>
        Public NamedTypeWithInitializers As Expression = Expr(Function() New Foo With {.Bar = "abcd", .Baz = "efgh"})

        <Category(NewObject)>
        Public NamedTypeConstructorParameters As Expression = Expr(Function() New Foo("ijkl"))

        <Category(NewObject)>
        Public NamedTypeConstructorParametersWithInitializers As Expression = Expr(Function() New Foo("ijkl") With {.Bar = "abcd", .Baz = "efgh"})

        <Category(NewObject)>
        Public AnonymousType As Expression = Expr(Function() New With {.Bar = "abcd", .Baz = "efgh"})

        <Category(NewObject)>
        Public AnonymousTypeFromVariables As Expression = IIFE(Function()
                                                                   Dim Bar = "abcd"
                                                                   Dim Baz = "efgh"
                                                                   Return Expr(Function() New With {Bar, Baz})
                                                               End Function)

        <Category(NewObject)>
        Public CollectionTypeWithInitializer As Expression = Expr(Function() New List(Of String) From {"abcd", "efgh"})

        <Category(NewObject)>
        Public CollectionTypeWithMultipleElementsInitializers As Expression = Expr(Function() New Wrapper From {{"ab", "cd"}, {"ef", "gh"}})

        <Category(NewObject)>
        Public CollectionTypeWithSingleOrMultipleElementsInitializers As Expression = Expr(Function() New Wrapper From {{"ab", "cd"}, "ef"})
    End Module
End Namespace